### ISSUE TITLE
Fix the implementation of htmlmerger.

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Starcounter#3498]
The first implementation of the htmlmerger was added as a hack in the jsonserializer. When multiple partial jsonresponses (siblings) are serialized an extra property is added during serialization which contain the apps and the html they use.

```javascript
{
...
  {
  "Launcher": { ... },
  "Products": { ... },
  "Html": "/htmlmerger?Launcher=launcherpage.html&Products=productspage.html"
...
}
```

Apart from being a temporary (and quite ugly) hack this stopped working in some cases after #3465 was implemented since the whole result is not necessarily sent. 

As a workaround, @warpech fixed how it works on the client (https://github.com/StarcounterPrefabs/Launcher/issues/224) with the effect that now it's required that the Launcher app to be running.

This functionality needs to be properly solved, so we can get rid of the hack and workaround. Also needed to be able to switch over to the new middleware handling for supplying html of partial jsonresponses.

This needs to be discussed, @tomalec, @warpech, @per-samuelsson. @warpech  added another issue with a possible solution for investigation, #3495

**Comment by chrhol** [comment]
Solution for this is in #3539. Will close this issue and add a new one that only contains the server-specific tasks.
